delira/utils/config.py

```diff
@@ -513,6 +515,135 @@ def create_from_str(cls, data, formatter=yaml.load, decoder_cls=Decoder,
                      **kwargs)
         return config
 
+    def create_argparser(self):
+        '''
+        Creates an argparser for all values in the config
+        Following the pattern: `--training.learning_rate 1234`
+
+        Returns
+        -------
+        argparse.ArgumentParser
+            parser for all variables in the config
+        '''
+        parser = argparse.ArgumentParser(allow_abbrev=False)
+
+        def str2bool(v):
+            if v.lower() in ('yes', 'true', 't', 'y', '1'):
+                return True
+            elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+                return False
+            else:
+                raise argparse.ArgumentTypeError('Boolean value expected.')
+
+        def add_val(dict_like, prefix=''):
```

Comment:
a general question here? Shouldn't we use the actual `val` as a default argument here (it the argument type is not a dict)? this would allow us to optionally overwrite these args but to also retain the same config if flags are not specified

Reply:
They will be retained as the old `Config` should be updated by the `argparser` output and not overwritten.